### PR TITLE
CHANGELOG: update changelog for 1.3.0 release

### PR DESCRIPTION
**Pull request by liangwen12year**
Signed-off-by: Wen Liang <liangwen12year@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,18 @@
 # Changelog
+## [1.2.1] - 2021-03-24
+### Changes
+- Use inclusive language
+Use inclusive language in docs, variable names, and more
```
Comment (tyll) [inline CHANGELOG.md:5]: This needs more explanation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,18 @@
 # Changelog
+## [1.2.1] - 2021-03-24
```
Comment (tyll) [inline CHANGELOG.md:2]: This should probably be 1.3.0 since there are new features and bigger changes for inclusive language

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,18 @@
 # Changelog
+## [1.3.0] - 2021-03-24
+### Changes
+- Use inclusive language
+Use inclusive language in docs, variable names, and more
+
+### New features
+- Support ipv6.method disabled (8318686)
+- Support dns-options (880b7ab)
```
Comment (tyll) [inline CHANGELOG.md:9]: Please explain which options and try to use more formal language instead of technical terms - also the actual setting would be with underscore and not with dashes. To me the commit IDs seem to be disturbing.

- Support to configure DNS options to ... and ...

Also noticed that the README does not explain which DNS options can be set and what they do and how they are written. Are these the same options that resolv.conf(5) describes? Then this could be written. If there are differences they should be explained.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+- Add support for ethtool coalesce settings (b586d11)
```
Comment (tyll) [inline CHANGELOG.md:10]:
```suggestion
- Ethtool coalesce settings support

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,18 @@
 # Changelog
+## [1.3.0] - 2021-03-24
+### Changes
```
Comment (ffmancera) [inline CHANGELOG.md:3]: Instead of changes, use `Breaking changes` and `Bug fixes`. For the inclusive language each parameter, should have attached its own description and should be on `Breaking Changes`

File: CHANGELOG.md (hunk 1 above)
Comment (ffmancera) [reply CHANGELOG.md:5]: Explain each parameter deprecated in favor of the new one.

**Review by ffmancera — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (tyll) [reply CHANGELOG.md:3]: FYI: the previous changelog entry used "Changes" not "Breaking Changes" since the role does not really break things at this time.

Comment (thom311) [reply CHANGELOG.md:3]: Right!! I almost wanted to comment "but the role doesn't break API, does it"?

File: CHANGELOG.md (hunk 3 above)
Comment (liangwen12year) [reply CHANGELOG.md:9]: They are the same DNS options as resolv.conf(5) describing

Comment (liangwen12year) [reply CHANGELOG.md:9]: I will create a new PR describing the available DNS options

File: CHANGELOG.md (hunk 5 above)
Comment (ffmancera) [reply CHANGELOG.md]: Well, IMO, deprecating something is a "breaking change" even if we are not dropping it in this version, it will be in the future.. But I am fine then to use "Changes" if the previous entry did.

File: CHANGELOG.md (hunk 3 above)
Comment (liangwen12year) [reply CHANGELOG.md:11]: But currently, we only support DNS options for **IPv4**

File: CHANGELOG.md (hunk 1 above)
Comment (liangwen12year) [reply CHANGELOG.md]: updated, thanks

File: CHANGELOG.md (hunk 5 above)
Comment (thom311) [reply CHANGELOG.md:3]: "deprecating" does not necessarily mean that it will be dropped in the future (https://en.wikipedia.org/wiki/Deprecation)

Taking deprecation as an excuse to break API later, is IMO bad (but common). We should not do that.

In any case, it would be a breaking change only in the future when it gets actually dropped.


Maybe one could say: but now I get a deprecation warning, and that already broke my setup. E.g. if libnm marks API as deprecated, you may get a compile time error with `-Wdeprecated-declarations=error`. Now one might argue whether that really counts as broken or still within acceptable limits. But if that is the concern, then the deprecation warning is too aggressive, and should be toned down. But not say: "well of course it broke, we depreacted it!".

File: CHANGELOG.md (hunk 3 above)
Comment (thom311) [reply CHANGELOG.md]: DNS should not be specific to the address family. NetworkManager gets this wrong with `ipv4.dns-options` and `ipv6.dns-options`.

How to correctly abstract that in the role, is a different question.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,38 @@
...
+- Use inclusive language
+Use inclusive language in docs, variable names, function names and parameter names
+  - "slave" is deprecated in favor of "port"
+  - "master" is deprecated in favor of "controller"
+
+### New features
+- Support ipv6.method disabled
+- Support dns_options for IPv4 config
+
+  The list of supported DNS options is:
```
Comment (tyll) [inline CHANGELOG.md:13]: This is too verbose for the changelog

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,38 @@
...
+## [1.3.0] - 2021-03-24
+### Changes
+- Use inclusive language
+Use inclusive language in docs, variable names, function names and parameter names
+  - "slave" is deprecated in favor of "port"
+  - "master" is deprecated in favor of "controller"
+
+### New features
+- Support ipv6.method disabled
+- Support dns_options for IPv4 config
```
Comment (tyll) [inline CHANGELOG.md:11]:
```suggestion
- Support dns_options when using one or more IPv4 nameservers
```

File: CHANGELOG.md (hunk 5 above)
Comment (tyll) [reply CHANGELOG.md:3]: Using deprecation is not an excuse but a warning so that users can start to migrate away from it.

File: CHANGELOG.md (hunk 6 above)
Comment (liangwen12year) [reply CHANGELOG.md:13]: How about now?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,36 @@
 # Changelog
+## [1.3.0] - 2021-03-24
+### Changes
+- Use inclusive language
+Use inclusive language in docs, variable names, function names and parameter names
```
Comment (tyll) [inline CHANGELOG.md:5]: This is repeating `Use inclusive language`, mentioning "docs, variable names, ..." is too much detail in my opition.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,36 @@
...
+  - "slave" is deprecated in favor of "port"
+  - "master" is deprecated in favor of "controller"
+
+### New features
+- Support ipv6.method disabled
```
Comment (tyll) [inline CHANGELOG.md:10]:
```suggestion
- Support disabling IPv6
```

File: CHANGELOG.md (hunk 6 above)
Comment (tyll) [reply CHANGELOG.md:13]: Please remove the list of options from here.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,4 +1,36 @@
...
+  - `no-reload`, disables automatic reloading of a changed configuration file.
+  - `no-tld-query`, attempt not to resolve an unqualified name as if it were a top level domain(TLD).
+  - `rotate`, round-robin selection of name servers from among those listed.
+  - `single-request`, this option makes glibc perform the IPv6 and IPv4 requests sequentially.
+  - `single-request-reopen`, open a new socket for the second request if two requests from same port are handled incorrectly.
+  - `timeout:n`, the amount of time the resolver will wait for a response from a remote name server.
+  - `trust-ad`, this option controls the AD bit behavior of the stub resolver.
+  - `use-vc`, this option forces the use of TCP for DNS resolutions.
+
+- Ethtool coalesce settings support
```
Comment (tyll) [inline CHANGELOG.md:28]:
```suggestion
- Support Ethtool coalesce settings
```

**Review by cathay4t — APPROVED** [review]

File: CHANGELOG.md (hunk 10 above)
Comment (liangwen12year) [reply CHANGELOG.md]: I changed into "Support Ethtool coalesce settings".

File: CHANGELOG.md (hunk 6 above)
Comment (liangwen12year) [reply CHANGELOG.md]: The list of DNS options was removed.

File: CHANGELOG.md (hunk 9 above)
Comment (liangwen12year) [reply CHANGELOG.md]: Thanks, I changed into "Support disabling IPv6".

File: CHANGELOG.md (hunk 8 above)
Comment (liangwen12year) [reply CHANGELOG.md:5]: I changed into 
```
- Use inclusive language
  - "slave" is deprecated in favor of "port"
  - "master" is deprecated in favor of "controller"
  ```

File: CHANGELOG.md (hunk 7 above)
Comment (liangwen12year) [reply CHANGELOG.md]: I changed into "Support dns_options when using one or more IPv4 nameservers".

**Comment by liangwen12year** [comment]
@cathay4t , rebased. Thanks.

**Comment by tyll** [comment]
This will probably need some fixes when #378 is merged.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,4 +1,19 @@
 # Changelog
+## [1.3.0] - 2021-03-24
+### Changes
+- Use inclusive language
+  - "slave" is deprecated in favor of "port"
+  - "master" is deprecated in favor of "controller"
```
Comment (tyll) [inline CHANGELOG.md:6]: Please use backticks for the technical terms instead of double quotes.

Comment (liangwen12year) [reply CHANGELOG.md]: I replaced double quotes with the backticks.

File: .mdl_style.rb [hunk 12]
```diff
@@ -5,3 +5,5 @@
 rule 'MD013', :line_length => 88
 # https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md029---ordered-list-item-prefix
 rule 'MD029', :style => :ordered
+# https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md024---multiple-headers-with-the-same-content
+rule "MD024", :allow_different_nesting => true
```
Comment (tyll) [inline .mdl_style.rb:9]: This should be its own commit with an explanatory commit message.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,6 +1,26 @@
 Changelog
 =========
 
+[1.3.0] - 2021-03-24
```
Comment (tyll) [inline CHANGELOG.md:4]: You will need to update the date here when the actual release is published. @ffmancera when is this planned?

File: .mdl_style.rb (hunk 12 above)
Comment (liangwen12year) [reply .mdl_style.rb:9]: I created a new commit for the above change

Comment (tyll) [reply .mdl_style.rb:9]: Please explain in the commit message why we need this change.

Comment (liangwen12year) [reply .mdl_style.rb:9]: I explained as following:
```
`markdownlint` check failed on CHANGELOG.md with error "Multiple
headers with the same content", because the `markdownlint` parameter
`allow_different_nesting` defaults to be `false`. To allow header
duplication under different nesting level, we need to explicitly set
`allow_different_nesting` to be true.
```

File: CHANGELOG.md (hunk 13 above)
Comment (ffmancera) [reply CHANGELOG.md:4]: In the moment it is approved. Maybe we can say today? :-)

**Review by ffmancera — APPROVED** [review]
LGTM, thanks!

File: .mdl_style.rb (hunk 12 above)
Comment (tyll) [reply .mdl_style.rb:9]: I would prefer an explanation that gives the technical reason why `allow_different_nesting=True` is necessary. Using the default value of `False` as reason is not a convincing reason. It is similar to argue that a function was added because it was not there in the beginning.

A possible alternative would be, to give you an example:

Markdownlint requires each section headers to be different by default but the Changelog needs to contain multiple sections with the same name, therefore enable `allow_different_nesting` to support this use case.

I will not block this PR on this, though.

File: CHANGELOG.md (hunk 13 above)
Comment (liangwen12year) [reply CHANGELOG.md]: I updated into `[1.3.0] - 2021-04-08`

Comment (liangwen12year) [reply CHANGELOG.md]: I updated into `[1.3.0] - 2021-04-08`

File: .mdl_style.rb (hunk 12 above)
Comment (liangwen12year) [reply .mdl_style.rb:9]: I updated with the examples that you provide, thanks.

**Comment by tyll** [comment]
#311 seems to be ready to be merged, too. So if it is merged before this one, the changelog needs to be updated.

**Review by tyll — APPROVED** [review]
LGTM
